Add ability to disable opflex-agent features

tbachman opened the pull request:
The opflex-agent is dependent on the vSwitch from the underling
distribution. Some distributions ship older versions of the vSwitch
which aren't capable of supporting new features in the opflex-agent,
and therefore those features have to be disabled. This patch adds
a new configuration item for this, which provides a string with a
comma-delimited list inside for the features that should be
disabled in the deployment.

The initial feature disabled is ERSPAN.

tbachman committed:
Add ability to disable opflex-agent features

The opflex-agent is dependent on the vSwitch from the underling
distribution. Some distributions ship older versions of the vSwitch
which aren't capable of supporting new features in the opflex-agent,
and therefore those features have to be disabled. This patch adds
a new configuration item for this, which provides a string with a
comma-delimited list inside for the features that should be
disabled in the deployment.

The initial feature disabled is ERSPAN.

tomflynn reviewed (commented):
I'm not really able to review this change as I'm not familiar with the syntax but if no features are to be disabled, how would you express that? Would you just omit the param and write an empty array in opflex-agent-ovs.conf

"feature": {
    "disabled" : []
}

We expect to have additional features that can be disabled in the future. In those cases we expect to get a comma separated set of strings

"feature": {
    "disabled" : ["erspan", "feature2"]
}

tbachman commented:
We'll need to make a different fix - this expects the commas to be within the string. I'll re-open the github issue, indicating that we need to fix it to support proper JSON.

tbachman commented:
Pushed an update, which addresses review comments. The comma_separated_list supports this type of construct (see https://docs.openstack.org/heat/latest/template_guide/hot_spec.html#hot-spec, and look for table with the data types, including comma_separated_list).
